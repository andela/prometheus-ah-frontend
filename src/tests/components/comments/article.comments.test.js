import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ConnectedArticleComments,
{ ArticleComments } from '../../../components/comment/ArticleComments';
import mockData from '../../__mocks__/commentMockData';

const props = {
  loadCommentsAction: jest.fn(),
  article: mockData.articles[0],
  comments: mockData.comments,
  auth: true,
  error: {},
  user: {
    username: 'faksam'
  },
  paginationMeta: mockData.paginationMeta,
  commentReply: {
    id: 2,
    createdAt: '2016-02-18T03:22:56.637Z',
    updatedAt: '2016-02-18T03:48:35.824Z',
    body: 'This article is hot',
    User: {
      username: 'joeeasy',
      email: 'joeeasy@gmail.com',
      bio: 'tomorrow is a better day',
      id: 1,
      isVerified: true,
      role: 'user'
    }
  },
};
const initialState = {
  commentsArray: [],
  commentPage: 1,
  articleReducer: {
    article: mockData.articles[0],
  },
  auth: {
    error: {},
    user: {
      username: 'joeeasy',
      email: 'joeeasy@gmail.com',
      bio: 'tomorrow is a better day',
      id: 1,
      isVerified: true,
      role: 'user'
    }
  },
  commentReducer: {
    comments: mockData.comments,
    paginationMeta: mockData.paginationMeta,
    commentReply: mockData.commentReplies[0],
    commentReplies: mockData.commentReplies,
  },
};
const mockStore = configureMockStore([thunk]);
const store = mockStore(initialState);

describe('ArticleComments Component', () => {
  it('should render without throwing an error', () => {
    const wrapper = shallow(<ArticleComments {...props} />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  it('should load more comments', () => {
    const wrapper = shallow(<ArticleComments {...props} />);
    const instance = wrapper.instance();
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.setState({ commentPage: 1 });
    instance.loadMoreComments(event);
    expect(props.loadCommentsAction).toHaveBeenCalled();
  });
  it('should set state to props', () => {
    const wrapper = shallow(<ConnectedArticleComments store={store} {...props} />);
    expect(wrapper.state()).toEqual({});
  });
});
