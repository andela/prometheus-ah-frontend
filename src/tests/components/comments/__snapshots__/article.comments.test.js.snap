// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleComments Component should render without throwing an error 1`] = `
<div>
  <br />
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h5>
        Comments Â 
        <span
          className="comments-count"
          id="commentsCount"
        />
      </h5>
      <div>
        <Connect(CommentForm)
          article={
            Object {
              "Tags": Array [],
              "User": Object {
                "bio": "tomorrow is a better day",
                "email": "joeeasy@gmail.com",
                "firstname": null,
                "id": 1,
                "image": null,
                "lastname": null,
                "role": "user",
                "status": "active",
                "username": "joeeasy",
              },
              "body": "PHP is a cool framework for coding but not fast as node",
              "createdAt": "2018-10-17T11:50:20.538Z",
              "description": "coding",
              "id": 11,
              "readingTime": "1 min read",
              "slug": "how-to-train-your-dragon",
              "title": "how to code",
              "updatedAt": "2018-10-17T11:50:20.538Z",
              "userId": 1,
            }
          }
        />
        <br />
        <br />
      </div>
      <Connect(CommentList)
        commentsArray={Array []}
        username="faksam"
      />
      <div>
        <button
          className="btn mx-auto"
          onClick={[Function]}
          type="submit"
        >
          Load More....
        </button>
      </div>
    </div>
  </div>
  <hr />
</div>
`;
